src/Adapter/MSTest.CoreAdapter/Discovery/TypeEnumerator.cs

```diff
@@ -106,11 +106,19 @@ internal UnitTestElement GetTestFromMethod(MethodInfo method, ICollection<string
             var isAsync = ReflectHelper.MatchReturnType(method, typeof(Task));
 
             var testMethod = new TestMethod(method.Name, this.type.FullName, this.assemblyName, isAsync);
+
             if (!method.DeclaringType.FullName.Equals(this.type.FullName))
             {
                 testMethod.DeclaringClassFullName = method.DeclaringType.FullName;
             }
 
+            if (!this.reflectHelper.IsMethodDeclaredInSameAssemblyAsType(method, this.type))
```

Comment:
Good idea. I was thinking since it's internal the method could be used outside `GetTests` but it's only called by unit tests, so OK.